pkg/docker/docker.go

```diff
@@ -609,15 +610,30 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 
 	containerName := containerNameOrID(container)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
+	removalDone := make(chan struct{}, 1)
+	removeContainer := func() {
 		glog.V(4).Infof("Removing container %q ...", containerName)
 		if err := d.RemoveContainer(container.ID); err != nil {
 			glog.Warningf("Failed to remove container %q: %v", containerName, err)
 		} else {
 			glog.V(4).Infof("Removed container %q", containerName)
 		}
+		removalDone <- struct{}{}
+	}
+	go func() {
+		sigchan := make(chan os.Signal, 1)
+		signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
```

Comment:
What's the rationale behind the choice of signals to listen to here?
It seems to be in agreement with the code in https://github.com/kubernetes/kubernetes/tree/master/pkg/util/interrupt (used in https://github.com/openshift/origin/pull/8306), that handles SIGINT, SIGTERM, SIGQUIT on Linux, and only SIGINT in other platforms.

By [default](https://golang.org/pkg/os/signal/#hdr-Default_behavior_of_signals_in_Go_programs), Go programs will panic on [synchronous signals](https://golang.org/pkg/os/signal/#hdr-Types_of_signals) (triggered by errors in program execution), and deferred calls will be executed: [example](https://gist.github.com/rhcarvalho/689ebc47aa39fd7069904e325ae43e06).
That means if our code panics for whatever reason, we should be covered.

Then, a SIGHUP, SIGINT, or SIGTERM signal causes the program to exit. In that case no deferred call is executed. Here we're handling SIGINT and SIGTERM, but not SIGHUP. Why?

SIGQUIT (and others), by default, cause the program to exit with a stack dump. By handling it here we're removing the default stack dump.

What is the signal that the platform (OpenShift/Kubernetes) send to terminate an on-going build?